### PR TITLE
Expose the VolumeHelper to third-party plugins.

### LINKED ISSUE #7818: third party BackupItemAction cannot support the VolumePolicy in v1.14.0
**Describe the problem/challenge you have**

https://github.com/vmware-tanzu/velero/blob/62c7fef8274260f6e5349197058666f01e12a60a/pkg/backup/item_backupper.go#L380-L397
Due to this logic check, the VolumePolicy is limited to the Velero CSI BIA, all the other BIA plugins cannot support the [VolumePolicy](https://velero.io/docs/main/resource-filtering/#resource-policies), including the [Velero vSphere plugin](https://github.com/vmware-tanzu/velero-plugin-for-vsphere).

This is a known issue for Velero v1.14.0. The reason is the current BackupItemAction mechanism doesn't have an easy way to let the Velero pass in the VolumePolicy information into the BIA.
The next patch version of v1.14 will give a way to address it.

**Describe the solution you'd like**

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by blackpiglet**
Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #7818 

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by kaovilai — APPROVED** [review]

File: pkg/util/volumehelper/volume_policy_helper.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+			crclient.ObjectKey{Namespace: backup.Namespace, Name: backup.Spec.ResourcePolicy.Name},
+			policiesConfigMap,
+		)
+		if err != nil {
+			logger.Errorf("Fail to get ResourcePolicies %s ConfigMap with error %s.",
+				backup.Namespace+"/"+backup.Spec.ResourcePolicy.Name, err.Error())
+			return false, fmt.Errorf("fail to get ResourcePolicies %s ConfigMap with error %s",
+				backup.Namespace+"/"+backup.Spec.ResourcePolicy.Name, err.Error())
+		}
+		resourcePolicy, err := resourcepolicies.GetResourcePoliciesFromConfig(policiesConfigMap)
```
Comment (reasonerjt) [inline pkg/util/volumehelper/volume_policy_helper.go:64]: It's a debt that sometimes it's singular (resourcePolicy...) and sometimes it's plural( resourcepolicies ...)

File: pkg/util/volumehelper/volume_policy_helper.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+func ShouldPerformSnapshotWithBackup(
+	unstructured runtime.Unstructured,
+	groupResource schema.GroupResource,
+	backup velerov1api.Backup,
+	crClient crclient.Client,
+	logger logrus.FieldLogger,
+) (bool, error) {
+	var resourcePolicy *resourcepolicies.Policies = nil
+
+	if backup.Spec.ResourcePolicy != nil &&
```
Comment (reasonerjt) [inline pkg/util/volumehelper/volume_policy_helper.go:50]: I suppose there should be other places initializing the ResoucePolicy of a backup, can we reuse it?

This func should only be called by plugin, right?  If the answer is true, IMHO we'd better put it in `pkg/plugin/util` or `pkg/util/plugin` to indicate we don't want it be called in velero or node agent.

File: pkg/util/volumehelper/volume_policy_helper.go (hunk 1 above)
Comment (blackpiglet) [reply pkg/util/volumehelper/volume_policy_helper.go]: Modified.

File: pkg/util/volumehelper/volume_policy_helper.go (hunk 2 above)
Comment (blackpiglet) [reply pkg/util/volumehelper/volume_policy_helper.go]: Modified.

File: internal/resourcepolicies/resource_policies.go [hunk 3]
```diff
@@ -148,7 +153,43 @@ func (p *Policies) Validate() error {
 	return nil
 }
 
-func GetResourcePoliciesFromConfig(cm *v1.ConfigMap) (*Policies, error) {
+func InitiateResourcePoliciesFromBackup(
```
Comment (reasonerjt) [inline internal/resourcepolicies/resource_policies.go:156]: It's trivial but I don't think the "Initiate" in the name makes a lot sense, why not just call it "GetResourcePoliciesFromBackup" or "ResourcePoliciesFromBackup"?

Comment (kaovilai) [reply internal/resourcepolicies/resource_policies.go:156]: I vote "GetResourcePoliciesFromBackup"

Comment (blackpiglet) [reply internal/resourcepolicies/resource_policies.go]: Renamed to `GetResourcePoliciesFromBackup`.

**Review by reasonerjt — APPROVED** [review]

**Review by Lyndon-Li — APPROVED** [review]

**Pull request merged**